apps/correlations/kinds/correlations.cue

```diff
@@ -0,0 +1,41 @@
+package kinds
+
+correlationsv0alpha1: {
+	kind:       "Correlation"  // note: must be uppercase
+	pluralName: "Correlations"
+	schema: {
+		spec: {
+			source_ds_ref:  DataSourceRef
+			target_ds_ref?:  DataSourceRef
+			label:       string
+			description?: string
+			config:      ConfigSpec
+			provisioned: bool
+			type:        CorrelationType
+		}
+	}
+}
+
+DataSourceRef: {
+   group: string // same as pluginId
+   name: string // same as grafana uid
+}
+
+
+// there was a deprecated field here called type, we will need to move that for conversion and provisioning
+ConfigSpec: {
+	field: string
+	target: TargetSpec
```

Comment:
what is this target?  is it related to `spec.target`?

Reply:
If the correlation type is query, `target` is the JSON for the query editor for the target datasource. See https://github.com/grafana/grafana/blob/main/devenv/datasources.yaml#L279 for a simple example, but the JSON the query editor expects will differ between datasources.

If the correlation type is external, `target` will look something like `{"url":"https://www.google.com/search?q=${go_gc_duration_seconds_sum}"`

Target will be an object, but the properties of the object will be different depending.